gaia-component.js

```diff
@@ -122,6 +54,14 @@ var base = {
         this.setInnerDirAttributes = setInnerDirAttributes.bind(null, this);
         document.addEventListener('dirchanged', this.setInnerDirAttributes);
       }
+      if (document.l10n) {
+        this.localize = document.l10n.translateFragment.bind(
+          document.l10n, this.shadowRoot);
+        document.l10n.ready.then(() => {
+          document.addEventListener('DOMRetranslated', this.localize);
+          this.localize();
+        });
+      }
```

Comment:
This could also give users that aren't using `document.l10n` a chance to localize shadow content by overriding the default behaviour. We could pass the `[data-l10n-id]` nodes to the callback so that users don't get direct access to the shadow-root. This would depend on use documenting all `[data-l10n-id]`s and regard changes to these as 'breaking'.

```js
XFoo.prototype.setupShadowL10n = function(nodes) {
  node.forEach(node => ...);
};
```

It's not the most elegant solution, but it does give non l10n.js users something :)